GHzDACs/ghz_fpga_server.py

```diff
@@ -1631,16 +1655,100 @@ def dac_run_sram(self, c, data, loop=False, blockDelay=0):
         """
         if not len(data):
             return
-        
-        if loop:
-            # make sure data is at least 20 words long by repeating it
-            data *= (20-1)/len(data) + 1
-        else:
-            # Set data at least 20 words long by repeating first value
+
+        if len(data) < 20:
+            # pad out to 20 with zeros
+            d2 = np.zeros(20)
+            d2[0:len(data)] = data
+            data = d2
             data = list(data) + [data[0]] * (20-len(data))
-        
+
         dev = self.selectedDAC(c)
         yield dev.runSram(data, loop, blockDelay)
+
+    @setting(2081, 'DAC Write SRAM', data='*w')
+    def dac_write_sram(self, c, data):
+        """ Write the given data to SRAM.
+
+        Note that if your board is currently running, it will immediately start
+        playing the new SRAM, wherever it is, so use with caution.
+        """
+        dev = self.selectedDAC(c)
+        yield dev._sendSRAM(np.array(data, dtype='<u4').tostring())
+
+    @setting(1082, "Add Jump Table Entry", name='s', arg=['ww{IDLE}', 'www{JUMP}', 'w{NOP,END}', 'wwww{CYCLE}'])
+    def add_jt_entry(self, c, name, arg=None):
```

Comment:
I prefer using name prefixes so that related settings will be grouped when sorting lexically; also, we should make the setting names and method names correspond (or, better yet, just make them identical by not providing the second parameter to the `@setting` decorator at all):
```python
@setting(1082, name='s',...)
def jump_table_add_entry(...):
...

def jump_table_clear(...):
...

def jump_table_set_counters(...):
...
```